### ISSUE TITLE
Profile `__init__()` missing keyword argument `async_persist` when calling `az login` through default cli `invoke()`

### ISSUE THREAD

**Issue by weibolu-rm** [Azure/azure-cli#20796]
## Describe the bug

**Command Name**
`az login`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
__init__() got an unexpected keyword argument 'async_persist'
Traceback (most recent call last):
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 658, in execute
    raise ex
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 692, in _run_job
    result = cmd_copy(params)
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/command_modules/profile/custom.py", line 106, in login
    profile = Profile(cli_ctx=cmd.cli_ctx, async_persist=False)
TypeError: __init__() got an unexpected keyword argument 'async_persist'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- use `get_default_cli()` to invoke `az login`
i.e. 

```
from azure.cli.core import get_default_cli

get_default_cli().invoke(['login'])
```

## Expected Behavior

successfully call `az login`

## Environment Summary
```
Linux-5.8.0-55-generic-x86_64-with-debian-bullseye-sid, Ubuntu 20.04.3 LTS
Python 3.6.10
Installer: DEB

azure-cli 2.31.0

```
## Additional Context
part of pip list:
```
azure-cli-command-modules-nspkg 2.0.3
azure-cli-core                  2.31.0
azure-cli-nspkg                 3.0.4
azure-cli-profile               2.1.5
azure-cli-telemetry             1.0.6
azure-common                    1.1.27
azure-core                      1.21.1
azure-identity                  1.7.1
azure-mgmt-core                 1.3.0
azure-mgmt-resource             18.0.0
azure-nspkg                     3.0.2
```

**Comment by jiasli** [comment]
Could be similar to https://github.com/Azure/azure-cli/issues/20297.

`async_persist` is from Azure CLI <= 2.29.0. Could you check `azure-cli` version by `pip list`? it should also be 2.31.0. Apparently the content of `/home/robert/dev/Project/venv/lib/python3.9/site-packages/azure/cli/command_modules/profile/custom.py:106` is incorrect:

https://github.com/Azure/azure-cli/blob/d7e5ede07748b423d2b03fbb8bd25b8bb2c19023/src/azure-cli/azure/cli/command_modules/profile/custom.py#L106

**Comment by weibolu-rm** [comment]
@jiasli you're right, we were relying on a system install of azure-cli, hence why we could only call the commands directly and not through our internal python package.

This is resolved by adding `azure-cli >= 2.31.0` as a requirement.

Thank you!
